src/components/common/PropTypes.ts

```diff
@@ -0,0 +1,6 @@
+export interface DynamicListProps<TModel> {
+  addText: string | ((listItems: Array<TModel>) => string);
+  modelFactory: () => TModel;
+  updateModel: (value: Array<TModel>) => void;
```

Comment:
onItemsUpdated? onListItemsUpdated?

Reply:
This, I like- it has the potential to be used for more than just updating the model, and I will include it in a new commit 😊